parser/src/powdr.lalrpop

```diff
@@ -72,7 +72,7 @@ Part: Part = {
 // Same as SymbolPath plus we allow "::<...>" at the end.
 GenericSymbolPath: (SymbolPath, Option<Vec<Type<Expression>>>) = {
     // If we "inline" SymbolPath here, we get an ambiguity error.
-    <abs:"::"?> <parts:( <Part> "::" )*> <end:Part> <types:("::" "<" <TypeTermList<Expression>> ">")?>  => (
+    <abs:"::"?> <parts:( <Part> "::" )*> <end:Part> <types:("::" "<" <TypeTermList<Expression<TermWithStruct>>> ">")?>  => (
```

Comment:
Does it make sense to define `Expression` as `Expression<TermWithStruct>` and use some other name for the one that takes the parameter? Like ExpressionRestricted or something?

Reply:
Can't have both `Expression` as a rename and `Expression<StructOption>` as a Macro. The only way would be something like `RestrictedExpression` and `FullExpression`, what do you think?